### PR TITLE
Add Request Model Support

### PR DESCRIPTION
**Pull request by beck3905**
*Issue #, if available:*
#895 and #931

*Description of changes:*
Added support for listing `Models` in `AWS::Serverless:API` and defining a model to be used in the `Api` event source.

*Description of how you validated changes:*
unit tests
end to end tests
ran transform locally and ran created a stack from the transformed template

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [x] Update documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by beck3905** [comment]
I'm not really sure what needs to be done to update documentation.  I couldn't see in the repo how the documentation gets generated in order to update it.

File: samtranslator/swagger/swagger.py [hunk 1]
```diff
@@ -37,6 +37,7 @@ def __init__(self, doc):
         self.paths = self._doc["paths"]
         self.security_definitions = self._doc.get("securityDefinitions", {})
         self.gateway_responses = self._doc.get(self._X_APIGW_GATEWAY_RESPONSES, {})
+        self.definitions = self._doc.get('definitions', {})
```
Comment (praneetap) [inline samtranslator/swagger/swagger.py]: To make this work with Openapi 3, you would need to move definitions under $/components/schemas.
You could add it as a post process similar to the auth components here - https://github.com/awslabs/serverless-application-model/blob/develop/samtranslator/model/api/api_generator.py#L293

Here is a link on whats changed from swagger 2.0 to openapi 3.0 - 
https://blog.readme.io/an-example-filled-guide-to-swagger-3-2/

Comment (beck3905) [reply samtranslator/swagger/swagger.py]: @praneetap I've updated the PR to include openapi support.  Will you please have a look?

Comment (praneetap) [reply samtranslator/swagger/swagger.py]: The changes look good to me, have you tried to deploy an openapi transformed template manually on cloud formation? 
We would also really appreciate it if you add an example to the [examples](https://github.com/awslabs/serverless-application-model/tree/master/examples/2016-10-31)/.

**Comment by beck3905** [comment]
@praneetap @keetonian I've added an example and update documentation as requested.

**Comment by beck3905** [comment]
@keetonian Thank you for the approval and merge.  What is the process once merged for new features to be released and generally available?

**Comment by keetonian** [comment]
@beck3905 we periodically pull in changes to the internal version of the transform, run a suite of integration tests, and gradually release the change through AWS regions. We post on corresponding GitHub issues once a change has been deployed in all regions and post release notes to this repo.

**Pull request merged**